standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java

```diff
@@ -352,26 +354,29 @@ public void alterTable(RawStore msdb, Warehouse wh, String catName, String dbnam
           String oldTblLocPath = srcPath.toUri().getPath();
           String newTblLocPath = dataWasMoved ? destPath.toUri().getPath() : null;
 
-          // also the location field in partition
-          parts = msdb.getPartitions(catName, dbname, name, -1);
-          for (Partition part : parts) {
-            String oldPartLoc = part.getSd().getLocation();
-            if (dataWasMoved && oldPartLoc.contains(oldTblLocPath)) {
-              URI oldUri = new Path(oldPartLoc).toUri();
-              String newPath = oldUri.getPath().replace(oldTblLocPath, newTblLocPath);
-              Path newPartLocPath = new Path(oldUri.getScheme(), oldUri.getAuthority(), newPath);
-              part.getSd().setLocation(newPartLocPath.toString());
-            }
-            part.setDbName(newDbName);
-            part.setTableName(newTblName);
-          }
           // Do not verify stats parameters on a partitioned table.
           msdb.alterTable(catName, dbname, name, newt, null);
           int partitionBatchSize = MetastoreConf.getIntVar(handler.getConf(),
               MetastoreConf.ConfVars.BATCH_RETRIEVE_MAX);
           String catalogName = catName;
           // alterPartition is only for changing the partition location in the table rename
           if (dataWasMoved) {
+            PartitionsRequest req = new PartitionsRequest(newDbName, newTblName);
```

Comment:
> Avoid calls get_partitions_req if not dataWasMoved

Please give more context about this change.

Reply:
When `dataWasMoved` is false, we do not need get all partitions and update the location of partitions.